### PR TITLE
docs(k4k8s) add changelog for Kong Enterprise k8s 1.4.2.0

### PR DESCRIPTION
**Pull request by hbagdi**

### CODE REVIEW

File: app/enterprise/k8s-changelog.md [hunk 1]
```diff
@@ -2,20 +2,42 @@
...
+
+Kong Enterprise k8s 1.4.2.0 is based on Kong Gateway 1.4.2,
+meaning it inherits all new features and changes between Kong Gateway 1.3.0 and
+1.4.2.
+Please checkout changelog for
+[Kong Gateway 1.4.0](https://github.com/Kong/kong/blob/master/CHANGELOG.md#140).
+
+### Enterprise plugins
+
+- `upstream-tls` plugin has been dropped. Users are advised to use
```
Comment (shashiranjan84) [inline app/enterprise/k8s-changelog.md]: It was never part of K4K8 release. 

 https://github.com/Kong/kong-distributions/blob/release/kic-1.3.0.0/kong-images/kic/distributions_constants.lua

Comment (h3xar0n) [reply app/enterprise/k8s-changelog.md]: Correct. It's mentioned in the main Enterprise 1.3.0.0 changelog:

https://github.com/Kong/docs.konghq.com/blame/master/app/enterprise/changelog.md#L306

File: app/enterprise/k8s-changelog.md [hunk 2]
```diff
@@ -2,20 +2,36 @@
...
+
+Kong Enterprise k8s 1.4.2.0 is based on Kong Gateway 1.4.2.
+
+### Kong Gateway
+
+Kong Enterprise k8s 1.4.2.0 is based on Kong Gateway 1.4.2,
+meaning it inherits all new features and changes between Kong Gateway 1.3.0 and
+1.4.2.
+Please checkout changelog for
+[Kong Gateway 1.4.0](https://github.com/Kong/kong/blob/master/CHANGELOG.md#140).
```
Comment (shashiranjan84) [inline app/enterprise/k8s-changelog.md]: is this correct?

Comment (hbagdi) [reply app/enterprise/k8s-changelog.md]: I'm not sure which part and what you are referring to here. Could you elaborate?

Comment (shashiranjan84) [reply app/enterprise/k8s-changelog.md]: the changlog pointing to 1.4.0, shouldn't it be [1.4.2](https://github.com/Kong/kong/blob/master/CHANGELOG.md#142)

Comment (hbagdi) [reply app/enterprise/k8s-changelog.md]: The reason I pointed it to 1.4.0 is that's the one that has most of the changes between 1.3.0 and 1.4.0. The two patch releases are just bug fixes and don't have anything notable there.

File: app/enterprise/k8s-changelog.md [hunk 3]
```diff
@@ -2,20 +2,36 @@
...
+### Summary
+
+Kong Enterprise k8s 1.4.2.0 is based on Kong Gateway 1.4.2.
+
+### Kong Gateway
+
+Kong Enterprise k8s 1.4.2.0 is based on Kong Gateway 1.4.2,
+meaning it inherits all new features and changes between Kong Gateway 1.3.0 and
+1.4.2.
+Please checkout changelog for
```
Comment (h3xar0n) [inline app/enterprise/k8s-changelog.md]:
```suggestion
Please review the changelog for
```

File: app/enterprise/k8s-changelog.md [hunk 4]
```diff
@@ -2,20 +2,36 @@
...
-Kong Enterprise k8s 1.3 is the debut release for this package.
+Kong Enterprise k8s 1.3.0.0 is the debut release for this package.
 
 ### Kong Gateway
 
-Kong Enterprise k8s 1.3 is based on Kong 1.3.0,
-meaning it inherits all the changes from it.
+Kong Enterprise k8s 1.3.0.0 is based on Kong Gateway 1.3.0,
+meaning it inherits all the features and changes from it.
 Please checkout changelog for
```
Comment (h3xar0n) [inline app/enterprise/k8s-changelog.md]:
```suggestion
Please review the changelog for
```

**Comment by hbagdi** [comment]
@h3xar0n Please merge when you are ready. Thanks!

**Pull request merged**